app/src/main/java/org/mozilla/focus/widget/FirefoxProgressBar.kt

```diff
@@ -0,0 +1,90 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+package org.mozilla.focus.widget
+
+import android.content.Context
+import android.util.AttributeSet
+import android.view.LayoutInflater
+import android.arch.lifecycle.Observer
+import android.os.Handler
+import android.os.Looper
+import android.os.Message
+import android.view.View
+import android.widget.LinearLayout
+import org.mozilla.focus.R
+import org.mozilla.focus.architecture.NonNullObserver
+import org.mozilla.focus.fragment.BrowserFragment
+import kotlinx.android.synthetic.main.firefox_progress_bar.view.*
+import java.lang.ref.WeakReference
+import java.util.concurrent.TimeUnit
+
+private const val HIDE_MESSAGE_ID = 0
+private const val HIDE_ANIMATION_DURATION_MILLIS = 250L
+private val HIDE_AFTER_MILLIS = TimeUnit.SECONDS.toMillis(3)
+
+class FirefoxProgressBar @JvmOverloads constructor(
+        context: Context, attrs: AttributeSet? = null, defStyle: Int = 0)
+    : LinearLayout(context, attrs, defStyle) {
+
+    private val hideHandler = FirefoxProgressBarHideHandler(this)
+
+    fun initialize(browserFrag: BrowserFragment) {
+        browserFrag.session.url.observe(browserFrag, Observer { sessionUrl -> url.text = sessionUrl })
+        browserFrag.session.loading.observe(browserFrag, object : NonNullObserver<Boolean>() {
+            public override fun onValueChanged(loading: Boolean) {
+                if (loading) {
+                    startUpdates()
+                } else {
+                    resetCountdown()
+                }
+            }
+        })
+    }
+
+    init {
+        LayoutInflater.from(context)
+                .inflate(R.layout.firefox_progress_bar, this, true)
+        gifImageView.setGifImageResource(R.drawable.progress_loading_indicator)
+
+        // hardware acceleration needs to be disabled for gif support in FirefoxProgressBar
+        // http://android-er.blogspot.it/2014/03/play-animated-gif-with.html
+        setLayerType(View.LAYER_TYPE_SOFTWARE, null)
```

Comment:
I'm uncomfortable with the potential side effects of this:
- Does drawing in software over a hardware layer force that hardware layer to draw in software too? We need to test this
- Using the CPU to draw animations will slow down pageload performance (which is using the CPU), especially on these slower devices

This person wrote a script to convert GIFs to AnimationDrawables: https://medium.com/@tushar.acharya/converting-animated-gifs-to-animationdrawables-c6594084d7ad Perhaps we should verify the scripts are sound and use that instead. So we can land this on live app testing, file a follow-up? @liuche What do you think?